src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchSyncSpec.groovy

```diff
@@ -354,6 +359,100 @@ class SwitchSyncSpec extends BaseSpecification {
+                it.meters.proper.find { it.meterId == broadcastCookieMeterId }.rate == meterToManipulate.rate
+            }
+        }
+    }
+
+    @Tidy
+    @Tags([VIRTUAL, SMOKE])
+    def "Able to synchronize misconfigured flow meter"() {
+        given: "An active switch with flow on it"
+        def sw = topology.activeSwitches.first()
+        def flow = flowHelperV2.singleSwitchFlow(sw)
+        flowHelperV2.addFlow(flow)
+
+        when: "Update flow's meter"
+        def flowMeterIdToManipulate = northbound.getAllMeters(sw.dpId).meterEntries.find {
+            it.meterId >= MIN_FLOW_METER_ID
+        }
+        def newBurstSize = flowMeterIdToManipulate.burstSize + 100
+        def newRate = flowMeterIdToManipulate.rate + 100
+        lockKeeper.updateBurstSizeAndRate(sw, flowMeterIdToManipulate.meterId, newBurstSize, newRate)
+
+        then: "Flow is not valid"
+        def responseValidateFlow = northbound.validateFlow(flow.flowId).findAll { !it.discrepancies.empty }*.discrepancies
+        assert responseValidateFlow.size() == 1
+        def meterRateDiscrepancies = responseValidateFlow[0].find { it.field.toString() == "meterRate" }
+        def meterBurstSizeDiscrepancies = responseValidateFlow[0].find { it.field.toString() == "meterBurstSize" }
+        meterRateDiscrepancies.actualValue == newRate.toString()
+        meterRateDiscrepancies.expectedValue == flowMeterIdToManipulate.rate.toString()
+        meterBurstSizeDiscrepancies.actualValue == newBurstSize.toString()
+        meterBurstSizeDiscrepancies.expectedValue == flowMeterIdToManipulate.burstSize.toString()
+
+        and: "Validate switch endpoint shows the updated meter as misconfigured"
+        Wrappers.wait(RULES_DELETION_TIME) {
+            def validateInfo = northbound.validateSwitch(sw.dpId)
+            assert validateInfo.meters.misconfigured.size() == 1
+            assert validateInfo.meters.misconfigured[0].actual.burstSize == newBurstSize
+            assert validateInfo.meters.misconfigured[0].expected.burstSize == flowMeterIdToManipulate.burstSize
+            assert validateInfo.meters.misconfigured[0].actual.rate == newRate
+            assert validateInfo.meters.misconfigured[0].expected.rate == flowMeterIdToManipulate.rate
+        }
+
+        when: "Synchronize switch"
+        northbound.synchronizeSwitch(sw.dpId, false)
+
+        then: "The misconfigured meter was fixed and moved to the 'proper' section"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            with(northbound.validateSwitch(sw.dpId)) {
+                it.meters.misconfigured.empty
+                it.meters.proper.find {
+                    it.meterId == flowMeterIdToManipulate.meterId
+                }.burstSize == flowMeterIdToManipulate.burstSize
+                it.meters.proper.find {
+                    it.meterId == flowMeterIdToManipulate.meterId
+                }.rate == flowMeterIdToManipulate.rate
+            }
+        }
```

Comment:
Is it fixed my re-installing meters + rules or by 'update' command?

Reply:
by 'update' command I guess.
I will sync with dev on this question

Reply:
I've synced with dev, it is fixed by 'updated'